lightningd/offer.c

```diff
@@ -14,17 +14,40 @@ static void json_populate_offer(struct json_stream *response,
 				const struct sha256 *offer_id,
 				const char *b12,
 				const struct json_escape *label,
+				const char *description,
 				enum offer_status status)
 {
 	json_add_sha256(response, "offer_id", offer_id);
 	json_add_bool(response, "active", offer_status_active(status));
 	json_add_bool(response, "single_use", offer_status_single(status));
 	json_add_string(response, "bolt12", b12);
+	if (description)
+		json_add_string(response, "description", description);
```

Comment:
A better approach is to use `json_add_stringn`, since it accepts an explicit length parameter and doesn’t require the string to be null-terminated. Using `json_add_string` will result in garbage data being appended after the `description` value.